File: output/ChatGPT/avatar/Java/C++/codeforces_190_A.cpp
#include <iostream>
using namespace std;

int main() {
    int n, m;
    cin >> n >> m;

    if (n == 0) {
        if (m != 0)
            cout << "Impossible" << endl;
        else
            cout << 0 << " " << 0 << endl;
        return 0;
    }

    if (m <= n) {
        if (m == 0)
            m = 1;
        cout << n << " " << (m + n - 1) << endl;
        return 0;
    }

    cout << m << " " << (m + n - 1) << endl;

}

File: output/ChatGPT/avatar/Python/Java/codeforces_190_A.java
int n, m;
Scanner scanner = new Scanner(System.in);
n = scanner.nextInt();
m = scanner.nextInt();

if (n < 1) {
    if (m < 1) {
        System.out.println(n + " " + m);
        System.exit(0);
    }
    System.out.println("Impossible");
    System.exit(0);
}

int min, max;

if (m > n) {
    min = m;
    max = n + m - 1;
    System.out.println(min + " " + max);
}

if (m <= n) {
    min = n;
    max = n + m - 1;
    if (m == 0) {
        max = n;
    }
    System.out.println(min + " " + max);
}